### PR TITLE
Add function to plot a model's incidence graph in Dulmage-Mendelsohn order

### PR DESCRIPTION
**Pull request by Robbybp**
## Summary/Motivation:
Thanks to a casual comment by @jeanpaulwatson, and because I've spent way too many hours reproducing these plots over and over, I'd like to add a method plot a model's incidence matrix in Dulmage-Mendelsohn order.

## Changes proposed in this PR:
- `spy_dulmage_mendelsohn` function that plots incidence matrix on Matplotlib Axes in Dulmage-Mendelsohn order.

I've tried to automatically compute a `markersize` that seems to work for both small and large-ish systems. For a small model, we get something like:
dm-auto-small

For a large-ish model, we get something like (with `include_wellconstrained=False`):
dm-auto-large

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://pyomo.readthedocs.io/en/stable/contribution_guide.html) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by Robbybp** [comment]
Windows coverage failure appears unrelated.

**Comment by mrmundt** [comment]
> Windows coverage failure appears unrelated.

Coverage failures are, 99.99999% of the time, unrelated. (And also obnoxious.)

File: pyomo/contrib/incidence_analysis/visualize.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+        Config options for ``IncidenceGraphInterface``
+
+    order: ``IncidenceOrder``, optional
+        Order in which to plot sparsity structure
+
+    highlight_coarse: bool, optional
+        Whether to draw a rectangle around the coarse partition
+
+    highlight_fine: bool, optional
+        Whether to draw a rectangle around the fine partition
```
Comment (blnicho) [inline pyomo/contrib/incidence_analysis/visualize.py:121]: These 3 arguments have defaults defined that are missing from the docstring.

**Review by blnicho — APPROVED** [review]
I have one minor docstring edit but otherwise this looks good!

**Pull request merged**